cmake/BLTInstallableMacros.cmake

```diff
@@ -428,37 +428,161 @@ endmacro(blt_add_target_compile_flags)
+    set(singleValuedArgs TARGET TLIST)
     set(multiValuedArgs)
 
-    ## parse the arguments to the macro
+    # parse the arguments to the macro
     cmake_parse_arguments(arg
          "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
 
+    # check for required arguments
     if(NOT DEFINED arg_TARGET)
-       message(FATAL_ERROR "TARGET is a required parameter for the blt_convert_to_system_includes macro.")
+        message(FATAL_ERROR "TARGET is a required parameter for the blt_find_target_dependencies macro")
     endif()
 
-    # PGI does not support -isystem
-    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
-        get_target_property(_include_dirs ${arg_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
-        # Don't copy if the target had no include directories
-        if(_include_dirs)
-            # Clear previous value in INTERFACE_INCLUDE_DIRECTORIES so it is not doubled
-            # by target_include_directories
-            set_property(TARGET ${arg_TARGET} PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
-            target_include_directories(${arg_TARGET} SYSTEM INTERFACE ${_include_dirs})
+    if(NOT DEFINED arg_TLIST OR NOT DEFINED ${arg_TLIST})
+        message(FATAL_ERROR "TLIST is a required parameter for the blt_find_target_dependencies macro")
+    endif()
+
+    set(_depends_on "")
+
+    # Get dependencies if BLT registered library
+    string(TOUPPER ${arg_TARGET} _target_upper)
+    if(_BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
+        list(APPEND _depends_on "${_BLT_${_target_upper}_DEPENDS_ON}")
+    endif()
+    unset(_target_upper)
+
+    # Get dependencies if CMake target
+    if(TARGET ${arg_TARGET})
+        get_property(_target_type TARGET ${arg_TARGET} PROPERTY TYPE)
+        if(NOT "${_target_type}" STREQUAL "INTERFACE_LIBRARY")
+            get_target_property(_propval ${arg_TARGET} LINK_LIBRARIES)
+            if(_propval)
+                list(APPEND _depends_on ${_propval})
+            endif()
+        endif()
+
+        # get interface link libraries
+        get_target_property(_propval ${arg_TARGET} INTERFACE_LINK_LIBRARIES)
+        if (_propval)
+            list(APPEND _depends_on ${_propval})
         endif()
     endif()
 
-    unset(_include_dirs)
-endmacro()
+    blt_list_remove_duplicates(TO _depends_on)
+    foreach(t ${_depends_on})
+        if (NOT "${t}" IN_LIST ${arg_TLIST})
+            list(APPEND ${arg_TLIST} ${t})
+            blt_find_target_dependencies(TARGET ${t} TLIST ${arg_TLIST})
+        endif()
+    endforeach()
+
+    unset(_depends_on)
+endmacro(blt_find_target_dependencies)
+
+
+##------------------------------------------------------------------------------
+## blt_convert_to_system_includes(TARGET   <target>
+##                                TARGETS  [<target>...]
+##                                CHILDREN [TRUE|FALSE]
+##                                [QUIET])
+##
+## Converts existing interface includes to system interface includes.
```

Comment:
I see that there was a discussion about not marking `TARGET` as deprecated for now. 

Can we still include a comment that `TARGET` will be deprecated in the future, so users should prefer the `TARGETS` keyword? As is (and without a comment), the duplication of `TARGET` and `TARGETS` is confusing.